apps/client/src/features/modals/sheets-modal/SheetsModal.tsx

```diff
@@ -0,0 +1,285 @@
+import { ChangeEvent, useEffect, useRef, useState } from 'react';
+import {
+  Alert,
+  AlertDescription,
+  AlertIcon,
+  AlertTitle,
+  Button,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Select,
+} from '@chakra-ui/react';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+import { OntimeRundown, ProjectData, UserFields } from 'ontime-types';
+
+import { PROJECT_DATA, RUNDOWN, SHEET_STATE, USERFIELDS } from '../../../common/api/apiConstants';
+import { maybeAxiosError } from '../../../common/api/apiUtils';
+import {
+  getSheetsAuthUrl,
+  getSheetState,
+  patchData,
+  postPreviewSheet,
+  postPushSheet,
+  uploadSheetClientFile,
+} from '../../../common/api/ontimeApi';
+import useSheetState from '../../../common/hooks-query/useSheetState';
+import { projectDataPlaceholder } from '../../../common/models/ProjectData';
+import { userFieldsPlaceholder } from '../../../common/models/UserFields';
+import ModalLink from '../ModalLink';
+import PreviewExcel from '../upload-modal/preview/PreviewExcel';
+
+import Step from './Step';
+
+interface SheetsModalProps {
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+export default function SheetsModal(props: SheetsModalProps) {
+  const { isOpen, onClose } = props;
+
+  const queryClient = useQueryClient();
+
+  const [rundown, setRundown] = useState<OntimeRundown | null>(null);
+  const [userFields, setUserFields] = useState<UserFields | null>(null);
+  const [project, setProject] = useState<ProjectData | null>(null);
+
+  const fileInputRef = useRef<HTMLInputElement>(null);
+  const sheetid = useRef<HTMLInputElement>(null);
+  const worksheet = useRef<HTMLSelectElement>(null);
+
+  const { data: sheetState, refetch } = useQuery({
```

Comment:
i'm trying to push the id and worksheet values through the Query but can't get it to work
is it possible or should it be done differently

Reply:
I am not sure I understand the question, lets chat!